51-android.rules

```diff
@@ -891,7 +904,7 @@ LABEL="cdc", ENV{adb_user}="yes", GOTO="android_usb_rule_match"
 LABEL="adbfast", ENV{adb_adb}="yes"
 LABEL="fast", ENV{adb_fast}="yes", GOTO="android_usb_rule_match"
 
-# ADB Debug and mass storage (note: android_ver<4.3)
+# ADB Debug and mass storage (note: generally android_ver<4.3; Moto Z3 Play when powered off)
 LABEL="adbmass", ENV{adb_adbmass}="yes"
```

Comment:
Oh, and I forgot to ask. Is this mass storage thing meant for this case too or should I make a separate LABEL?

Reply:
I read earlier that mass storage was removed due to MS [Read stackoverflow](https://android.stackexchange.com/questions/190138/how-to-use-usb-mass-storage-mode-on-android-4-3), and [reddit](https://www.reddit.com/r/Android/comments/mg14z/comment/c30q93p/?rdt=52391), but there is also MTP and PTP, which isn't the same as mass storage, so it may be worth keeping them separate.

Reply:
Well, what I meant here is that this phone (running Android 9) can still present itself as an actual USB mass storage device. This doesn't happen under Android tho. There, only MTP and PTP are available as far as file sharing protocols are concerned. When I plug a turned off phone into my computer, `/dev/sdb` appears:
```
Jul 14 02:10:30 andrad kernel: usb 1-1: new high-speed USB device number 12 using xhci_hcd
Jul 14 02:10:30 andrad kernel: usb 1-1: New USB device found, idVendor=22b8, idProduct=2eb7, bcdDevice= 4.04
Jul 14 02:10:30 andrad kernel: usb 1-1: New USB device strings: Mfr=1, Product=2, SerialNumber=3
Jul 14 02:10:30 andrad kernel: usb 1-1: SerialNumber: <serialno>
Jul 14 02:10:31 andrad kernel: usb-storage 1-1:1.0: USB Mass Storage device detected
Jul 14 02:10:31 andrad kernel: scsi host1: usb-storage 1-1:1.0
Jul 14 02:10:32 andrad kernel: scsi 1:0:0:0: Direct-Access     Linux    File-Stor Gadget 0404 PQ: 0 ANSI: 2
Jul 14 02:10:32 andrad kernel: sd 1:0:0:0: Power-on or device reset occurred
Jul 14 02:10:32 andrad kernel: sd 1:0:0:0: [sdb] Media removed, stopped polling
Jul 14 02:10:32 andrad kernel: sd 1:0:0:0: [sdb] Attached SCSI removable disk
```
However, this is likely a very different thing from what Android <4.3 was doing. And so, maybe it should be handled by a separate code path.

Reply:
Interesting info about the /dev/sdb during power-off.

Reply:
Indeed. I'm quite perplexed what it would be for. At first I assumed it would be provide something like Windows drivers, but as far as I remember I couldn't mount it – it was behaving like empty SD card readers do (when connected by USB).